src/Groups/perm.jl

```diff
@@ -387,3 +387,168 @@ function cycle_structure(g::PermGroupElem)
+
+
+################################################################################
+#
+#   perm
+#
+@doc Markdown.doc"""
+    @perm(ex)
+Macro to input a permutation as
+`pi = @perm (1,2,3)(4,5)(6,7,8)` to obtain
+the permutation `(1,2,3)(4,5)(6,7,8)`, that is, the output of
+`cperm([1,2,3],[4,5],[6,7,8])`.
+# Examples
+```jldoctest
+julia> @perm (1,2,3)(4,5)(6,7,8)
+(1,2,3)(4,5)(6,7,8)
+```
+"""
+macro perm(ex)
+    res = []
+    
+    if typeof(ex) != Expr
+        error("Input is not a permutation.")
+    end
+    
+    if ex.head != :call && ex.head != :tuple
+        error("Input is not a permutation.")
+    end
+   
+    while ex.head == :call
+        pushfirst!(res, Expr(:vect, ex.args[2:end]...))
+        ex = ex.args[1]
+        if typeof(ex) != Expr
+            error("Input is not a permutation.")
+        end
+        if ex.head != :call && ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+    end
+    
+    if ex.head != :tuple
+        error("Input is not a permutation.")
+    end
+    
+    pushfirst!(res, Expr(:vect,ex.args...))
+
+    return esc(:(Oscar.cperm($(res...))))
+end
+
+
+################################################################################
+#
+#   perm(n,gens)
+#
+@doc Markdown.doc"""
+    @perm(n,gens)
+Macro to input a list of permutations which are generated as elements of
+the `symmetric_group(n)` with the function `cperm`.
+# Examples
+```jldoctest
+julia> gens = Oscar.Permutations.@perm 14 [
+               (1,10)
+              (2,11)
+              (3,12)
+              (4,13)
+              (5,14)
+              (6,8)
+              (7,9)
+              (1,2,3,4,5,6,7)(8,9,10,11,12,13,14)
+              (1,2)(10,11)
+             ]
+9-element Vector{PermGroupElem}:
+ (1,10)
+ (2,11)
+ (3,12)
+ (4,13)
+ (5,14)
+ (6,8)
+ (7,9)
+ (1,2,3,4,5,6,7)(8,9,10,11,12,13,14)
+ (1,2)(10,11)
+julia> gens[1].parent
+Sym( [ 1 .. 14 ] )
+```
+"""
+macro perm(n,gens)
+
+    s = symmetric_group(n)
+    ores = Vector{Expr}(undef,length(gens.args))
+    i = 1
+    for ex in gens.args
+        res = []
+        
+        if typeof(ex) != Expr
+            throw(ArgumentError("Input is not a permutation."))
+            error("Input is not a permutation.")
+        end
+        
+        if ex.head != :call && ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+       
+        while ex.head == :call
+            pushfirst!(res, Expr(:vect, ex.args[2:end]...))
+            ex = ex.args[1]
+            if typeof(ex) != Expr
+                error("Input is not a permutation.")
+            end
+            if ex.head != :call && ex.head != :tuple
+                error("Input is not a permutation.")
+            end
+        end
+        
+        if ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+        
+        pushfirst!(res, Expr(:vect,ex.args...))
+
+        ores[i] = esc(:(Oscar.cperm(symmetric_group($n),$(res...))))
```

Comment:
This calls `symmetric_group` again and again, which isn't great, as we create a new group object each time. This could be improved by using `let` block; but I don't want to complicate things even further, so I suggest we first iron out everything else and get this merged, then we can worry about fine tuning.

Just for the record, here is what I'd do: I'd only collect the `res` values here:
```suggestion
        ores[i] = esc(:(  [$(res...)] ))
```

Then, the final `return` of this macro could look something like this:
```julia
return quote
   let g = symmetric_group($n)
       [ cperm(g, pi) for pi in [$(ores...)] ]
   end
end
```
Note: normally we'd have to worry that `g` and `pi` could clash with symbols defined by the programmer in the surrounding code, and referenced in the permutation expression.  This is where the `esc` function comes into play: we applied it to the expressions we got from the user, but *not* to the outer expression above. This leads to the Julia [macro hygiene](https://docs.julialang.org/en/v1/manual/metaprogramming/#Hygiene) functionality kicking in, which determines that `g` and `pi` (unlike `symmetric_group` and `cperm`) are not known variables, and replaces them by special generated names. You can check the effect of this via `macroexpand`:
```
julia> g = 7  # introduce a variable g and use it in the macro expression:
7

julia> macroexpand(Main, :( Oscar.@perm 14 [
                     (g,9)
                     (1,2)(10,11)
                    ]))
quote
    let var"#332#g" = Oscar.symmetric_group(14)
        [Oscar.cperm(var"#332#g", var"#333#pi"...) for var"#333#pi" = [[[g, 9]], [[1, 2], [10, 11]]]]
    end
end
```